cranelift/codegen/src/dominator_tree.rs

```diff
@@ -580,9 +580,37 @@ impl DominatorTreePreorder {
     pub fn dominates(&self, a: Block, b: Block) -> bool {
         let na = &self.nodes[a];
         let nb = &self.nodes[b];
+
+        // A block is considered to dominate itself
+        if a == b {
+            return true;
+        }
```

Comment:
I'm surprised you needed to add this case -- it should be implied by the below (`pre_number` and `post_number` will both be equal pairwise to the other nodes' values if `a == b`). Is this a performance optimization or...?